src/v/iceberg/manifest_io.h

```diff
@@ -0,0 +1,89 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+#pragma once
+
+#include "base/outcome.h"
+#include "cloud_io/remote.h"
+#include "iceberg/manifest.h"
+#include "model/fundamental.h"
+#include "utils/named_type.h"
+
+#include <seastar/core/future.hh>
+
+#include <system_error>
+
+namespace iceberg {
+
+using manifest_path
+  = named_type<std::filesystem::path, struct manifest_path_tag>;
+
+class manifest_io {
+public:
+    enum class errc {
+        // The call has timed out. Callers should retry, if allowed.
+        timedout,
+
+        // The call failed for some reason that doesn't appear to be retriable.
+        failed,
+
+        // The call failed because the system is shutting down.
+        shutting_down,
+    };
+    explicit manifest_io(
+      cloud_io::remote& io, cloud_storage_clients::bucket_name b)
+      : io_(io)
+      , bucket_(b) {}
+    ~manifest_io() = default;
+
+    ss::future<checked<manifest, errc>>
```

Comment:
what is the rationale behind using `checked` instead of `result`?

Reply:
`manifest` doesn't have a copy constructor, which IIUC is why it's failing to build for me when I use a `result`